packages/zealot/src/query/result-stream.ts

```diff
@@ -87,7 +87,7 @@ export class ResultStream {
           (e instanceof Error && e.message.match(/context canceled/))
         ) {
           this.status = "aborted"
-          resolve()
+          reject(e)
```

Comment:
This fixes the first error you found. When aborting, it was still resulting in success. Not anymore.